base-image/Dockerfile

```diff
@@ -12,32 +14,22 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# The tel2-base target is reused by all telepresence cluster-side components. It's
-# never published though, as it's quite large and all builds will access to this Dockerfile
-FROM golang:alpine3.15 as tel2-base
+FROM golang:alpine3.15 as tel2-build
```

Comment:
Cool stuff going on here with the cache! I think it breaks the smart agent image generation process though. As you can see in here https://github.com/datawire/telepresence2-proprietary/blob/master/Makefile#L43 the smart agent will first run `make tel2-base`, which will build the docker image specifying `tel2-base` as a target. I'd expect a PR in `telepresence2-proprietary` to fix that dependency.
I started looking into building multi-arch images and that dependency to tel2-base from the smart agent build was a problem for me, so I'm looking forward to this new solution!